core/transaction_pool/impl/transaction_pool_impl.hpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_TRANSACTION_POOL_IMPL_HPP
+#define KAGOME_TRANSACTION_POOL_IMPL_HPP
+
+#include <outcome/outcome.hpp>
+#include "common/logger.hpp"
+#include "transaction_pool/pool_moderator.hpp"
+#include "transaction_pool/transaction_pool.hpp"
+
+namespace kagome::transaction_pool {
+
+  struct ReadyTransaction : public primitives::Transaction {
+    std::list<common::Hash256> unlocks;
+  };
+
+  struct WaitingTransaction : public primitives::Transaction {};
+
+  class TransactionPoolImpl : public TransactionPool {
+    static constexpr auto kDefaultLoggerTag = "Transaction Pool: ";
+    static constexpr size_t kDefaultMaxReadyNum = 512;
+    static constexpr size_t kDefaultMaxWaitingNum = 128;
+
+   public:
+    template <template <typename T> typename Container>
+    static auto create(
+        std::unique_ptr<PoolModerator> moderator,
+        Limits limits = Limits{kDefaultMaxReadyNum, kDefaultMaxWaitingNum},
+        common::Logger logger = common::createLogger(kDefaultLoggerTag)) {
+      return std::make_shared<TransactionPoolImpl>(
+          TransactionPoolImpl{std::move(moderator), limits, std::move(logger)});
```

Comment:
Why is it static factory method, and not in constructor?
Why is it templated at all?

Reply:
Static factories for components (e.g. impl of interface) works bad with DI, so better use constructor